syscalls.c

```diff
@@ -264,7 +263,7 @@ int syscalls_beginthreadex(void *ustack)
 	void (*start)(void *);
 	unsigned int priority, stacksz;
 	void *stack, *arg;
-	unsigned int *id;
+	handle_t *id;
```

Comment:
tid is not a handle_t, should be plain int
```suggestion
	int *id;
```